docs/docs/adrs/adr-018-permissionless-ics.md

```diff
@@ -0,0 +1,270 @@
+Note that `consumerID` could just be a `uint64` but we choose to include `chainID` in it as well so that it is easier to see
+what the consumer chain just by looking at the `consumerID`. This means that the `chainID` of a chain cannot be changed
+after launching it (because the `chainID` is part of the `consumerID` key).
+
+We intend to set a fixed cost of a `MsgLaunchConsumerChain` to avoid getting spammed with bogus consumer launch chains.
+
+To execute a `MsgLaunchConsumerChain`, we first create a `ConsumerAdditionProposal` under the hoods, with the `top_N` set to 0, and call
+[`HandleConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L30)
+simulating a proposal that was already voted on and accepted. Note that we need to migrate `ConsumerAdditionProposal`s to use
+the `consumerID` instead using [both](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/keeper/proposal.go#L382)
+`chainID` and the `spawnTime` as keys. The [usual validity conditions](https://github.com/cosmos/interchain-security/blob/v4.3.0/x/ccv/provider/types/proposal.go#L114)
+hold for the fields of `MsgLaunchConsumerChain`. Note however, that we intend to have a `spawnTime` upper limit as well.
+For example, if you launch a consumer chain in Permissionless ICS, the `spawnTime` should not be more
+than two months ahead in the future, to avoid having consumer chains lingering for too long before they get added. To do this,
+we introduce a `maxSpawnTime` limit in [`ConsumerAdditionProposal`](https://github.com/cosmos/interchain-security/blob/v4.3.0/proto/interchain_security/ccv/provider/v1/provider.proto#L29).
+This way, Opt In consumer chains that reach their `spawnTime` but have no validator opted in, get simply removed.
+
+#### Modify a Consumer Chain
+
+We introduce the `MsgUpdateConsumerChain` message so that the owner of a consumer chain can change its parameters
+(e.g., `spawn_time`, PSS-related parameters, etc.) This message can only be executed by the owner of a consumer chain (see `owner_address` in the `ConsumerChainRecord`).
+
+```protobuf
+message MsgUpdateConsumerChain {
+  // for which chain to change the state ...
+  string consumer_id;
+
+  // if the user intends to change the owner_address
+  // the one issuing this message should be able to sign this message as well?
+  string owner_address;
+  
+  // the chain-id of the new consumer chain
+  string chain_id;
+  // the proposed initial height of new consumer chain.
+  ibc.core.client.v1.Height initial_height;
+
+  ...  (all fields contained in ConsumerAdditionProposal except top_N)
+  
+  uint32 validators_power_cap;
+  uint32 validator_set_cap;
+  repeated string allowlist;
+  repeated string denylist;
+}
+```
+
+The `owner_address` is provided as well and hence a validator can change the owner of a consumer chain as well. Because of this,
```

Comment:
Why `a validator`? This has nothing to do with validators, and it should just say "the current owner account"?

Reply:
Wanted to write a consumer chain instead of a validator.